docs/src/modules/spring/pages/views.adoc

```diff
@@ -1 +1,472 @@
 = Implementing Views
+
+include::ROOT:partial$include.adoc[]
+
+You can access a single https://docs.kalix.io/reference/glossary.html#entity[Entity] with its https://docs.kalix.io/reference/glossary.html#entity_key[Entity key]. But sometimes this is not enough. You might want to retrieve multiple Entities, or retrieve them using an attribute other than the key. Kalix https://docs.kalix.io/reference/glossary.html#view[Views] allow you achieve this. You can create each View, so each one is optimized for an specific query.
+
+Views can be defined from any of the following:
+
+* xref:value-entity[Value Entities state changes]
+* xref:event-sourced-entity[Event Sourced Entity events]
+* xref:topic-view[Messages received from subscribing to topics on a broker]
+
+The remainder of this page describes:
+
+* <<transform-results>>
+* <<#changing>>
+* <<#query>>
+
+IMPORTANT: Be aware that Views are not updated immediately when Entity state changes. Kalix does update Views as quickly as possible, but it is not instant and can take up to a few seconds for the changes to become visible in the query results. View updates might also take more time during failure scenarios than during normal operation.
+
+[#value-entity]
+== Creating a View from a Value Entity
+
+Consider an example of a Customer Registry service with a `customer` Value Entity. When `customer` state changes, the entire state is emitted as a value change. Value changes update any associated Views. To create a View that lists customers by their name:
+
+* <<_define_the_view>> for a service that selects customers by name and associates a table name with the View. The table is created and used by Kalix to store the View.
+
+
+This example assumes the following `customer` exists
+
+[source,java,indent=0]
+.src/main/java/customer/api/Customer.java
+----
+include::example$spring-customer-registry-views-quickstart/src/main/java/customer/api/Customer.java[tags=class]
+----
+
+As well as a Value Entity service `CustomerEntity.java` that will produce the state changes consumed by the View:
+
+
+=== Define the View
+
+To get a View of multiple customers by their name, you need the set the following:
+
+ - Define the name of the table with `@Table`.
+ - Subscribe to the Value Entity with `@Subscribe.ValueEntity`.
+ - Extend from the class `View`.
+ - Define an endpoint for the view with `@GetMapping`. Is this optional? Can the view be called by a component?
```

Comment:
No, it's not optional. The goal of a View is to expose some representation of the model. The REST annotation is mandatory. 

If user wants to block it they need to add a restrictive ACL to this method.

Actions can call Views.

Reply:
thanks for the clarification. Deleted